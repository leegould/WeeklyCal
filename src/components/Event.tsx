import React from 'react';
import { TextInput, View, Text, StyleSheet, Switch, Animated, KeyboardAvoidingView, TouchableOpacity, Alert } from 'react-native';
import moment, { Moment } from 'moment';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { Button, Icon } from 'react-native-elements';
import { CalendarEvent } from '../types';
import DateTimeButton from './DateTimeButton';

export interface Props {
    navigation: {
        getParam: Function,
        goBack: Function,
    },
    onAddEvent: Function,
    onEditEvent: Function,
    onDeleteEvent: Function,
};

interface State { 
    allDay: boolean;
    anim: Animated.Value;
}

interface Values {
    title: string;
    startDate: Moment;
    endDate: Moment;
    // allDay: boolean; // This doesn't work for updating the form when the value changes
};

const ValidationSchema = Yup.object().shape({
    title: Yup.string()
        .min(2, 'Too Short!')
        .max(100, 'Too Long!')
        .required('Required'),
    startDate: Yup.date().required('Required'),
    endDate: Yup.date().when('startDate', (st: Date) => {
        return Yup.date().min(st, 'Must be after start!');
    }),
});

export default class Add extends React.PureComponent<Props, State> {
    static navigationOptions = {
        title: 'Add Event',
    };

    constructor(props: Props) {
        super(props);
        
        const event = (this.props.navigation.getParam('event', '') as CalendarEvent);
        const allDay = event ? event.allDay : true;

        this.state = {
            allDay,
            anim: new Animated.Value(allDay ? 0 : 1),
        };
    }

    toggleAllDay = () => this.setState((previousState: State) => {
        if (!previousState.allDay) {
            this.hide();
        } else {
            this.show();
        }
        return {
            allDay: !previousState.allDay,
        }
    });

    show() {
        Animated.timing(this.state.anim, {
            toValue: 1,
            duration: 250,
        }).start()
    }

    hide() {
        Animated.timing(this.state.anim, {
            toValue: 0,
            duration: 250,
        }).start()
    }

    render() {
        const date = (this.props.navigation.getParam('date', moment()) as Moment).hours(10).minute(0).second(0);
        const event = (this.props.navigation.getParam('event', '') as CalendarEvent);
        const existingId = event.id;
        const allowsUpdate = event.calendar ? event.calendar.allowsModifications : true;

        return (
            <KeyboardAvoidingView style={{flex: 1, backgroundColor: '#00000080'}} behavior='padding'>
                <TouchableOpacity 
                    style={{flex: 1}}
                    onPressOut={() => {this.props.navigation.goBack()}}
                >
                    <Formik
                        initialValues={{
                            title: event.title ? event.title : '',
                            startDate: event.startDate ? moment(event.startDate) : moment(date).hours(10).minute(0).second(0),
                            endDate: event.endDate ? moment(event.endDate) : moment(date).hours(16).minute(0).second(0),
                            // allDay: true,
                        }}
                        validationSchema={ValidationSchema}
                        onSubmit={(values: Values, formikActions) => {
                            console.log('values', values);
                            setTimeout(() => {
                                const event: CalendarEvent = {
                                    title: values.title,
                                    startDate: values.startDate.toDate(),
                                    allDay: this.state.allDay,
                                    endDate: undefined,
                                };

                                if (!this.state.allDay) {
                                    event.allDay = false;
                                    event.endDate = values.endDate.toDate();
                                }

                                if (existingId) {
                                    event.id = existingId;
                                    this.props.onEditEvent(event);
                                } else {
                                    this.props.onAddEvent(event);
                                }
                                
                                formikActions.setSubmitting(false); // turn off disabled

                                this.props.navigation.goBack();
                            }, 500);
                        }}
                    >
                        {props => (
                            <View style={styles.container}>
                                <View style={styles.subcontainer}>
                                    <View style={styles.rowContainer}>
                                        <Text style={styles.text}>All Day</Text>
                                        <Switch
                                            onValueChange={this.toggleAllDay}
                                            value={this.state.allDay}
                                            trackColor={{true: '#C2272D', false: ''}}
                                            style={styles.switchInput}
                                        />
                                    </View>
                                    <View style={styles.rowContainer}>
                                        <Text style={styles.text}>{this.state.allDay ? 'Date' : 'From'}</Text>
                                        <DateTimeButton
                                            showTime={this.state.allDay}
                                            date={props.values.startDate}
                                            onDateChanged={(date: Moment) => {
                                                props.handleChange('startDate');
                                                props.values.startDate = date;
                                            }}
                                        />
                                    </View>
                                    {props.touched.startDate && props.errors.startDate ?
                                    <Text style={styles.errorText} >
                                        {props.errors.startDate}
                                    </Text>
                                    : null }
                                    <Animated.View style={
                                        [styles.rowContainer,
                                            {
                                                opacity: this.state.anim,
                                                height: this.state.anim.interpolate({
                                                    inputRange: [0, 1],
                                                    outputRange: [0, 40]
                                                })
                                            }
                                        ]}
                                    >
                                        <Text style={styles.text}>To</Text>
                                        <DateTimeButton
                                            showTime={this.state.allDay}
                                            date={props.values.endDate}
                                            onDateChanged={(date: Moment) => {
                                                props.handleChange('endDate');
                                                props.values.endDate = date;
                                            }}
                                        />
                                    </Animated.View>
                                    {!this.state.allDay && props.touched.endDate && props.errors.endDate ?
                                    <Text style={styles.errorText} >
                                        {props.errors.endDate}
                                    </Text>
                                    : null }
                                    <TextInput
                                        style={styles.textInput}
                                        onChangeText={props.handleChange('title')}
                                        onBlur={props.handleBlur('title')}
                                        value={props.values.title}
                                        placeholder="Title"
                                        autoFocus
                                    />
                                    {props.touched.title && props.errors.title ?
                                    <Text style={styles.errorText} >
                                        {props.errors.title}
                                    </Text>
                                    : null }
                                    <View style={styles.rowButtons}>
                                        {existingId && allowsUpdate &&
                                        <Button
                                            onPress={() => {
                                                console.log('onDeletePress', event);
                                                Alert.alert('Delete Event', 'Are you sure you want to delete this event?',
                                                [
                                                    {text: 'No', onPress: () => { console.log('cancelled'); }, style: 'cancel'},
                                                    {text: 'Yes', onPress: () => {
                                                        this.props.navigation.goBack();
                                                        this.props.onDeleteEvent(event);
                                                    }},
                                                ])
                                            }}
                                            title='Delete'
                                            titleStyle={styles.buttonTitle}
                                            icon={<Icon name='delete-forever' type='material-community' color='#C2272D' size={20} />}                                   
                                            type='solid'
                                            buttonStyle={styles.button}
                                        />
                                        }
                                        <Button
                                            onPress={() => this.props.navigation.goBack()}
                                            title='Cancel'
                                            titleStyle={styles.buttonTitle}
                                            icon={<Icon name='close-circle' type='material-community' color='#C2272D' size={20} />}                                   
                                            type='solid'
                                            buttonStyle={styles.button}
                                        />
                                        {allowsUpdate &&
                                        <Button
                                            onPress={props.handleSubmit as any}
                                            title={existingId ? 'Save' : 'Add'}
                                            disabled={props.isSubmitting}
                                            titleStyle={styles.buttonTitle}
                                            icon={<Icon name='calendar-plus' type='material-community' color='#C2272D' size={20} />}
                                            type='solid'
                                            buttonStyle={styles.button}
                                        />
                                        }
                                    </View>
                                </View>
                            </View>
                        )}
                    </Formik>
                </TouchableOpacity>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        marginHorizontal: 40
    },
    subcontainer: {
        backgroundColor: 'lightgray',
        paddingBottom: 5,
        borderRadius: 5,
        paddingTop: 5,
    },
    text: {
        color:'white'
    },
    errorText: {
        color: '#C2272D',
    },
    checkboxContainer: {
        backgroundColor: 'lightgray',
        padding: 0,
        justifyContent: 'flex-end',
        borderColor: 'lightgray'
    },
    rowButtons: {
        flexDirection: 'row',
        justifyContent: 'space-evenly',
        alignItems: 'center',
    },
    rowContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginLeft: 5,
        height: 40,
    },
    switchInput: {
        transform: [{ scaleX: .8 }, { scaleY: .8 }]
    },
    textInput: {
        color: 'darkgray',
        height: 40,
        backgroundColor: 'beige',
        padding: 5,
    },
    buttonTitle: {
        marginLeft: 5,
        color: 'white',
        fontSize: 16
    },
    button: {
        backgroundColor: 'lightgray',
        borderWidth: 0
    },
});
