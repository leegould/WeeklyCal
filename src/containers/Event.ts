import { connect } from 'react-redux';
import Add from '../components/Event';
import { addEvent } from '../actions';
import { SimpleCalendarEvent } from '../types';

const mapDispatchToProps = (dispatch: any) => ({
    onAddEvent: (event: SimpleCalendarEvent) => {
      dispatch(addEvent(event));
    },
});

export default connect(null, mapDispatchToProps)(Add);