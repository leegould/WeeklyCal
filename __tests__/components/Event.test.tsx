import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import { Event } from '../../src/components/Event'; // Get the one without the mapped props.
import { CalendarEvent } from '../../src/types';

jest.useFakeTimers();

test('renders correctly with expected args', () => {
    const props = {
        navigation: {
            getParam: () => {},
            goBack: () => {},
        },
        onAddEvent: () => {},
        onEditEvent: () => {},
        onDeleteEvent: () => {},
        date: moment('25/12/2005', 'DD/MM/YYYY'),
        event: {
            // id: '123',
            startDate: moment('25/12/2005', 'DD/MM/YYYY').toDate(),
            endDate: moment('25/12/2005', 'DD/MM/YYYY').toDate(),
            allDay: true,
            title: 'test',
        } as CalendarEvent,
    }

    const tree = renderer.create(
        <Event 
            navigation={props.navigation}
            onAddEvent={props.onAddEvent}
            onEditEvent={props.onEditEvent}
            onDeleteEvent={props.onDeleteEvent}
            date={props.date}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
