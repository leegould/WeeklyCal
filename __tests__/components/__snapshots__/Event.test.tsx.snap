// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with expected args 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#00000080",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 40,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightgray",
            "borderRadius": 5,
            "paddingBottom": 5,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
                "marginLeft": 5,
              },
              Object {
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#C2272D",
                "fontSize": 18,
              }
            }
          >
            Add Event
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
              "marginLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            All Day
          </Text>
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor=""
            value={true}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
              "marginLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            Date
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "lightgray",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#C2272D",
                              "fontSize": 18,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        ïƒ­
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "marginLeft": 5,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    25-12-2005
                  </Text>
                </View>
              </View>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              onRequestClose={[Function]}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "black",
                    "bottom": 0,
                    "height": 1334,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 750,
                  }
                }
              />
              <View
                hideModalContentWhileAnimating={false}
                onModalHide={[Function]}
                onModalWillHide={[Function]}
                onModalWillShow={[Function]}
                pointerEvents="box-none"
                scrollOffset={0}
                scrollOffsetMax={0}
                scrollTo={null}
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                    "margin": 10,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
                supportedOrientations={
                  Array [
                    "portrait",
                    "landscape",
                  ]
                }
                swipeThreshold={100}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 13,
                        "marginBottom": 8,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderBottomColor": "#d5d5d5",
                        "borderBottomWidth": 0.5,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#8f8f8f",
                            "fontSize": 13,
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Pick a date
                    </Text>
                  </View>
                  <View
                    onStartShouldSetResponderCapture={[Function]}
                  >
                    <View>
                      <RCTDatePicker
                        date={1135504800000}
                        minuteInterval={1}
                        mode="date"
                        onChange={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "height": 216,
                          }
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#d5d5d5",
                        "borderTopWidth": 0.5,
                        "height": 57,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#007ff9",
                            "fontSize": 20,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 13,
                      "height": 57,
                      "justifyContent": "center",
                      "marginBottom": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#007ff9",
                          "fontSize": 20,
                          "fontWeight": "600",
                          "padding": 10,
                          "textAlign": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 0,
              "justifyContent": "space-between",
              "marginLeft": 5,
              "opacity": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            To
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "lightgray",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#C2272D",
                              "fontSize": 18,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        ïƒ­
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "marginLeft": 5,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    25-12-2005
                  </Text>
                </View>
              </View>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              onRequestClose={[Function]}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "black",
                    "bottom": 0,
                    "height": 1334,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 750,
                  }
                }
              />
              <View
                hideModalContentWhileAnimating={false}
                onModalHide={[Function]}
                onModalWillHide={[Function]}
                onModalWillShow={[Function]}
                pointerEvents="box-none"
                scrollOffset={0}
                scrollOffsetMax={0}
                scrollTo={null}
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                    "margin": 10,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
                supportedOrientations={
                  Array [
                    "portrait",
                    "landscape",
                  ]
                }
                swipeThreshold={100}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 13,
                        "marginBottom": 8,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderBottomColor": "#d5d5d5",
                        "borderBottomWidth": 0.5,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#8f8f8f",
                            "fontSize": 13,
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Pick a date
                    </Text>
                  </View>
                  <View
                    onStartShouldSetResponderCapture={[Function]}
                  >
                    <View>
                      <RCTDatePicker
                        date={1135526400000}
                        minuteInterval={1}
                        mode="date"
                        onChange={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "height": 216,
                          }
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#d5d5d5",
                        "borderTopWidth": 0.5,
                        "height": 57,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#007ff9",
                            "fontSize": 20,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 13,
                      "height": 57,
                      "justifyContent": "center",
                      "marginBottom": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#007ff9",
                          "fontSize": 20,
                          "fontWeight": "600",
                          "padding": 10,
                          "textAlign": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Title"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "beige",
              "color": "darkgray",
              "height": 40,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "lightgray",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#C2272D",
                            "fontSize": 20,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      ï…™
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 16,
                      "marginLeft": 5,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "lightgray",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#C2272D",
                            "fontSize": 20,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      ïƒ³
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 16,
                      "marginLeft": 5,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
